tests/datasets/test_querylog_processor.py

```diff
@@ -245,3 +245,111 @@ def test_missing_fields() -> None:
             ],
             None,
         )
+
+
+def test_error_code() -> None:
```

Comment:
this test seems larger than the change, is it testing for more than just the error code?

Reply:
No, I copied the `simple` test and added the error code.

Reply:
I see, can this be refactored to introduce less code duplication? Asking because from this test it is not clear that what's being tested is the error code. Either the scope of the test should change or the simple test part should be refactored to be reusable

Reply:
I collapsed it into the simple test.